core/src/test/java/kafka/test/junit/DetectThreadLeak.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package kafka.test.junit;
+
+import org.apache.kafka.test.TestUtils;
+
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+public class DetectThreadLeak {
```

Comment:
The purpose of this class is accurate, so maybe we can simplify it to a `FunctionalInterface`. For example:

```java
interface DetectThreadLeak {
    // return the new threads after `DetectThreadLeak` is created
    List<Thread> newThreads();
}
```

Also, we can create `DetectThreadLeak` with filter
```java
DetectThreadLeak.filter(Predicator<Thread>);
```

Reply:
I added `DetectThreadLeak` interface. Thanks.